packages/binding-file/src/file-client.ts

```diff
@@ -17,51 +17,39 @@
  * File protocol binding
  */
 import { Form, SecurityScheme } from "@node-wot/td-tools";
-import { ProtocolClient, Content, createLoggers } from "@node-wot/core";
+import { ProtocolClient, Content, createLoggers, ContentSerdes } from "@node-wot/core";
 import { Subscription } from "rxjs/Subscription";
 import fs = require("fs");
-import path = require("path");
+import { fileURLToPath } from "node:url";
 
-const { debug, warn } = createLoggers("binding-file", "file-client");
+const { debug } = createLoggers("binding-file", "file-client");
 
 export default class FileClient implements ProtocolClient {
     public toString(): string {
         return "[FileClient]";
     }
 
-    public async readResource(form: Form): Promise<Content> {
-        const filepath = form.href.split("//");
-        const resource = fs.createReadStream(filepath[1]);
-        const extension = path.extname(filepath[1]);
-        debug(`FileClient found '${extension}' extension`);
-        let contentType;
-        if (form.contentType != null) {
-            contentType = form.contentType;
-        } else {
-            // *guess* contentType based on file extension
-            contentType = "application/octet-stream";
-            switch (extension) {
-                case ".txt":
-                case ".log":
-                case ".ini":
-                case ".cfg":
-                    contentType = "text/plain";
-                    break;
-                case ".json":
-                    contentType = "application/json";
-                    break;
-                case ".jsonld":
-                    contentType = "application/ld+json";
-                    break;
-                default:
-                    warn(`FileClient cannot determine media type of '${form.href}'`);
-            }
-        }
+    private async readFromFile(filePath: string, contentType: string) {
+        debug(`Reading file of Content-Type ${contentType} from path ${filePath}.`);
+        const resource = fs.createReadStream(filePath);
         return new Content(contentType, resource);
     }
 
+    public async readResource(form: Form): Promise<Content> {
+        const filePath = fileURLToPath(form.href);
+        const contentType = form.contentType ?? ContentSerdes.DEFAULT;
+
+        return this.readFromFile(filePath, contentType);
+    }
+
     public async writeResource(form: Form, content: Content): Promise<void> {
-        throw new Error("FileClient does not implement write");
+        const filePath = fileURLToPath(form.href);
+        content.toBuffer();
```

Comment:
needed?

Reply:
Thank you, good catch :) Fixed by b6a364a7e10885f1e550070a966d6e222dd96d74